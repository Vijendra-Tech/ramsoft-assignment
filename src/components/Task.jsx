import React, { useEffect } from "react";
import { useSelector } from "react-redux";

function Task() {
  const tasks = useSelector((state) => state.tasks);
  return (
    <div>
      {tasks?.map((task) => (
        <div key={task.name} className="task">
          <p>{task.name}</p>
          <h6>{task.description}</h6>
          <h6>{task.ETA}</h6>
          <h6>{task.status}</h6>
        </div>
      ))}
    </div>
  );
}

export default Task;
