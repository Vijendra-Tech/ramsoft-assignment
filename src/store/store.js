import { configureStore } from "@reduxjs/toolkit";
import { taskSlice } from "../action/taskSlice";

const store = configureStore({
  reducer: { tasks: taskSlice.reducer },
});

export default store;
