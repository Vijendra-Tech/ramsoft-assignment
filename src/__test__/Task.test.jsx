import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";

import Task from "../components/Task";
import { Provider } from "react-redux";
import store from "../store/store";

describe("task Component", () => {
  test("Task component should render", () => {
    const task = render(
      <Provider store={store}>
        <Task />
      </Provider>
    );
    expect(task).toMatchSnapshot();
    task.unmount();
  });
});
