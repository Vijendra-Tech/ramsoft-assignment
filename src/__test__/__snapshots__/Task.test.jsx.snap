// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`task Component > Task component should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="task"
        >
          <p>
            Dummy
          </p>
          <h6>
            Dummy Decsc
          </h6>
          <h6>
            Mon Aug 14 2023 13:05:22 GMT+0530 (India Standard Time)
          </h6>
          <h6>
            Todo
          </h6>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="task"
      >
        <p>
          Dummy
        </p>
        <h6>
          Dummy Decsc
        </h6>
        <h6>
          Mon Aug 14 2023 13:05:22 GMT+0530 (India Standard Time)
        </h6>
        <h6>
          Todo
        </h6>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
